gdal/frmts/wms/wmsdriver.cpp

```diff
@@ -146,14 +146,44 @@ CPLXMLNode * GDALWMSDatasetGetConfigFromURL(GDALOpenInfo *poOpenInfo)
     }
 
     if (osSRSValue.empty())
+    {
         osSRSValue = "EPSG:4326";
 
-    if (osBBOX.empty())
+        if (osBBOX.empty())
+        {
+            if (osBBOXOrder.compare("yxYX") == 0)
+                osBBOX = "-90,-180,90,180";
+            else
+                osBBOX = "-180,-90,180,90";
+        }
+    }
+    else
     {
-        if (osBBOXOrder.compare("yxYX") == 0)
-            osBBOX = "-90,-180,90,180";
-        else
-            osBBOX = "-180,-90,180,90";
+        if (osBBOX.empty()) {
+            OGRSpatialReference oSRS;
+            oSRS.SetFromUserInput(osSRSValue);
+            oSRS.AutoIdentifyEPSG();
+
+            double dfWestLongitudeDeg, dfSouthLatitudeDeg,
+                    dfEastLongitudeDeg, dfNorthLatitudeDeg;
+            oSRS.GetAreaOfUse(&dfWestLongitudeDeg, &dfSouthLatitudeDeg,
+                    &dfEastLongitudeDeg, &dfNorthLatitudeDeg, nullptr);
+            OGRCoordinateTransformation *poCT = 
```

Comment:
auto poCT = std::unique_ptr<OGRCoordinateTransformation>(...)
and check return value